### PR TITLE
Tests: Clean up registered post types

### PR DESCRIPTION
**Pull request by obenland**
There are tests that register custom post types but don't clean them up after. This results in an unexpected menu structure for the admin menu tests that I'm currently working on in #17629.

Before change: https://travis-ci.com/github/Automattic/jetpack/jobs/430331383
After change: https://travis-ci.com/github/Automattic/jetpack/jobs/430390368

See #17629.

#### Changes proposed in this Pull Request:

* Adds `unregister_post_type` calls to every test that registers a post type.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

To run unit tests:
* Follow [setup guidelines](https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#unit-testing) if you don't have unit tests set up yet.
* Run `phpunit`

#### Proposed changelog entry for your changes:

* None needed.

### CODE REVIEW

**Review by kraftbj — DISMISSED** [review]
Thank you for this!

The failures are unrelated to this PR, so Crew will need to figure that out.

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 1]
```diff
@@ -630,6 +630,9 @@ function test_filters_out_blacklisted_post_types() {
 		$this->assertFalse( $this->server_replica_storage->get_post( $post_id ) );
 		$sync_event = $this->server_event_storage->get_most_recent_event( 'jetpack_sync_save_post' );
 		$this->assertFalse( $sync_event );
+
+		// Clean up.
+		unregister_post_type( 'snitch' );
```
Comment (mdbitz) [inline tests/php/sync/test_class.jetpack-sync-posts.php:635]: we actually need to move the unregister_post_type before the assertions. Otherwise if this test fails it will still be registered and disrupt other tests.

**Review by mdbitz — CHANGES_REQUESTED** [review]
Thanks @obenland for these updates. We definitely do need to cleanup these unit tests. However we also need to account for failing cases. 

In all of the modified tests the unregister actions need to be moved before the assertions that way if the tests fail it still gets unregistered and won't impact the other tests you are working on.

**Comment by obenland** [comment]
Nice catch @mdbitz! Patch is updated

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 2]
```diff
@@ -993,9 +1011,11 @@ function test_do_not_sync_non_public_post_types_filtered_post_content() {
 		$this->sender->do_sync();
 		$synced_post = $this->server_replica_storage->get_post( $post_id );
 
-		$this->assertEquals( '', $synced_post->post_content_filtered );
-		$this->assertEquals( '', $synced_post->post_excerpt_filtered );
+		// Clean up.
+		unregister_post_type( 'non_public' );
 
+		$this->assertSame( '', $synced_post->post_content_filtered );
```
Comment (mdbitz) [inline tests/php/sync/test_class.jetpack-sync-posts.php:1017]: @obenland do you mind outlining why the change to **assertSame**?

assertSame -> Asserts that two objects refer to the same object

Comment (obenland) [reply tests/php/sync/test_class.jetpack-sync-posts.php:1017]: PHPCS wouldn't let me commit with `assertEquals` 🤷

Comment (mdbitz) [reply tests/php/sync/test_class.jetpack-sync-posts.php:1017]: interesting. checking the branch out to see what it is yelling about :) Thanks for taking the initiative to clean-up these tests we greatly appreciate it.

Comment (mdbitz) [reply tests/php/sync/test_class.jetpack-sync-posts.php:1017]: Interesting, so assertSame is preferred over assertEquals 🎩 another good addition of today I learned.

> assertEquals accepts many non-string values, please use strict alternatives like assertSame

**Review by mdbitz — APPROVED** [review]
Approved! Thanks again :)

File: tests/php/sync/test_class.jetpack-sync-posts.php (hunk 1 above)
Comment (anomiex) [reply tests/php/sync/test_class.jetpack-sync-posts.php]: I'm late for this patch, but another pattern for that sort of thing is to do your code like
```
register_post_type( 'snitch', $args );
try {
	// All your code in here
} finally {
	// Clean up.
	unregister_post_type( 'snitch' );
}
```
That'll also do the right thing if some other code you call throws unexpectedly.

Comment (obenland) [reply tests/php/sync/test_class.jetpack-sync-posts.php]: Since then I also found [`reset_post_types()`](https://github.com/WordPress/wordpress-develop/blob/d6fa3df42031c493ce6c4b4ff5f32be1745067d1/tests/phpunit/includes/abstract-testcase.php#L263), which Core runs for its tests on `setUp`.
